src/main/java/software/amazon/lambda/snapstart/CacheLambdaHandlerFields.java

```diff
@@ -0,0 +1,42 @@
+package software.amazon.lambda.snapstart;
+
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.ba.XClass;
+import org.apache.bcel.classfile.Field;
+
+import java.util.HashSet;
+
+/**
+ * This detector stores fields with the Lambda Handler and Crac resources to be used later
+ * for visiting the classes passed in through dependency injection
+ */
+public class CacheLambdaHandlerFields implements Detector {
+
+    public static HashSet<String> fieldsToVisit = new HashSet<>();
```

Comment:
I added a comment where we use this too, but this variable should probably just be private, and we need a method to access these values instead.


I'm not 100% sure also about using a static variable to hold this (I don't know how SpotBugs manages this and if it creates an instance of this class or what). Maybe we could change it in the future if we understand this a bit more (Which is a good reason to create a method instead of accessing the variable directly)

Reply:
Framework facilitates this use case, see ReturnValueRandomnessPropertyDatabase.